src/binary.ts

```diff
@@ -37,51 +44,45 @@ export class Binary {
   /** User BSON type */
   static readonly SUBTYPE_USER_DEFINED = 128;
 
-  buffer: BinarySequence;
+  buffer: Buffer;
   sub_type: number;
   position: number;
 
   /**
    * @param buffer - a buffer object containing the binary data.
    * @param subType - the option binary type.
    */
-  constructor(buffer: string | BinarySequence, subType?: number) {
+  constructor(buffer?: string | BinarySequence, subType?: number) {
     if (
-      buffer != null &&
+      !(buffer === undefined || buffer === null) &&
       !(typeof buffer === 'string') &&
-      !Buffer.isBuffer(buffer) &&
-      !(buffer instanceof Uint8Array) &&
+      !ArrayBuffer.isView(buffer) &&
```

Comment:
`ArrayBuffer.isView()` returns true for any typed array class, this includes Buffer since it is a subclass of Uint8Array, as well a DataView. ensureBuffer now handles bytes-ifying any typed array / view